beetsplug/musicbrainz.py

```diff
@@ -115,18 +95,93 @@ def get_message(self):
     "recording-rels",
     "release-rels",
 ]
-if "work-level-rels" in musicbrainzngs.VALID_BROWSE_INCLUDES["recording"]:
-    BROWSE_INCLUDES.append("work-level-rels")
 BROWSE_CHUNKSIZE = 100
 BROWSE_MAXTRACKS = 500
 
 
+class LimiterTimeoutSession(LimiterMixin, TimeoutSession):
+    pass
+
+
+@dataclass
+class MusicBrainzAPI(RequestHandler):
+    session_type = LimiterTimeoutSession
+
+    api_host: str
+    rate_limit: float
+
+    @cached_property
+    def session(self) -> LimiterTimeoutSession:
+        return self.session_type(per_second=self.rate_limit)
+
+    def get_entity(self, entity: str, **kwargs) -> JSONDict:
+        return self.group_relations(
+            self.get_json(
+                f"{self.api_host}/ws/2/{entity}",
+                params={**kwargs, "fmt": "json"},
+            )
+        )
+
+    def get_release(self, id_: str) -> JSONDict:
+        return self.get_entity(f"release/{id_}", inc=" ".join(RELEASE_INCLUDES))
+
+    @singledispatchmethod
+    @classmethod
+    def group_relations(cls, data: Any) -> Any:
```

Comment:
I dont think the correct typehints are created for this, right?
```
    @overload
    @classmethod
    def group_relations(cls, data: List[T]) -> List[T]: ...
    
    @overload
    @classmethod
    def group_relations(cls, data: JSONDict) -> JSONDict: ...
```

Reply:
Are you talking about `data: Any` specifically?

Reply:
I will want to revisit this once I finish typing this module fully 😈